.buildkite/pipeline.py

```diff
@@ -0,0 +1,319 @@
+        msg = tm.render(pipeline=self)
+        return msg
+
+@dataclass(unsafe_hash=True)
+class Group:
+    """Buildkite Group object"""
+
+    project: str
+    category: str
+    steps: list[str]
+
+    def __lt__(self, other):
+        return self.project < other.project
+
+    def create_entity(self):
+        data = """
+{% if group.steps|length > 0 %}
+  - group: "{{ group.project }} {{ group.category }}"
+    key: "{{ group.project }}-{{ group.category }}"
+    steps:
+      {% for step in group.steps|sort -%}
+        {{ step.create_entity() }}
+      {% endfor -%}
+{% endif -%}
+"""
+
+        tm = Template(data)
+        msg = tm.render(group=self)
+        return msg
+
+@dataclass(unsafe_hash=True)
+class Agent:
+    """Buildkite Agent object"""
+    
+    image: str
+
+    def create_entity(self):
+        raise NotImplementedError("Not implemented yet")
+    
+@dataclass(unsafe_hash=True)
+class AWSAgent(Agent):
+    """AWS Agent object"""
+    
+    image: str 
+
+    def create_entity(self):
+        data = """
+        agents:
+          provider: "aws"
+          imagePrefix: "{{ agent.image }}"         
+          instanceType: "t4g.large"
+"""
+
+        tm = Template(data)
+        msg = tm.render(agent=self)
+        return msg
+
+@dataclass(unsafe_hash=True)
+class GCPAgent(Agent):
+    """GCP Agent object"""
+    
+    image: str 
+
+    def create_entity(self):
+        data = """
+        agents:
+          provider: "gcp"
+          image: "{{ agent.image }}"                   
+"""
+
+        tm = Template(data)
+        msg = tm.render(agent=self)
+        return msg
+@dataclass(unsafe_hash=True)
+class OrkaAgent(Agent):
+    """Orka Agent object"""
+    
+    image: str 
+
+    def create_entity(self):
+        data = """
+        agents:
+          provider: "orka"
+          imagePrefix: "{{ agent.image }}"                   
+"""
+
+        tm = Template(data)
+        msg = tm.render(agent=self)
+        return msg
+
+@dataclass(unsafe_hash=True)
+class Step:
+    """Buildkite Step object"""
+
+    command: str
+    agent: Agent
+    name: str
+    project: str
+    category: str
+    label: str = field(init=False)
+    comment: str = field(init=False)
+
+    def __post_init__(self):
+        self.comment = "/test " + self.project + " " + self.name
+        self.label = self.name
+
+    def __lt__(self, other):
+        return self.name < other.name
+
+    def create_entity(self):
+        data = """
+      - label: "{{ step.project }} {{ step.name }}"
+        command:
+          - "{{ step.command }}"
+        notify:
+          - github_commit_status:
+              context: "{{ step.project }}: {{ step.name }}"
+        {{ step.agent.create_entity() }}
+"""
+
+        tm = Template(data)
+        msg = tm.render(step=self)
+        return msg
+
+
+# Conditions:
+
+def is_pr() -> bool:
+    pr = os.getenv('BUILDKITE_PULL_REQUEST')
+    return pr and os.getenv('BUILDKITE_PULL_REQUEST') != "false"
```

Comment:
can the logic in the above two lines be simplified to `os.getenv('BUILDKITE_PULL_REQUEST') != "false"` ?